internal/providers/github/manager/manager.go

```diff
@@ -282,3 +283,57 @@ type credentialDetails struct {
 	ownerFilter sql.NullString
 	isOrg       bool
 }
+
+func (g *githubProviderManager) EncryptedCredentials(ctx context.Context, cred v1.Credential, state string) (*crypto.EncryptedData, error) {
```

Comment:
I think this method of the provider manager might be misplaced and misnamed. I started with this implementation but I wonder if better would be to provide instead a method on the credentials to return them encrypted or just use the crypto API directly and instead let the providerManager method just verify the credential which in the github case would call verifyCredentialsFromState